packages/frontend/src/providers/ExplorerProvider.tsx

```diff
@@ -981,9 +982,13 @@ function reducer(
                         tableCalculations:
                             state.unsavedChartVersion.metricQuery.tableCalculations.map(
                                 (tableCalculation) =>
-                                    tableCalculation.name ===
+                                    getItemId(tableCalculation) ===
                                     action.payload.oldName
-                                        ? action.payload.tableCalculation
+                                        ? {
+                                              ...action.payload
+                                                  .tableCalculation,
+                                              ...tableCalculation,
```

Comment:
I think it should be 

```
{

                                              ...tableCalculation,
                                              ...action.payload
                                                  .tableCalculation,
}
```

Otherwise, you can't update the tablecalculation 

Perhaps this is why I can't update now my TC formula or name. 

[Screencast from 21-11-23 13:20:40.webm](https://github.com/lightdash/lightdash/assets/1983672/8659144f-c2ca-412f-baef-9fce120c1463)